base/bitset.jl

```diff
@@ -123,6 +122,84 @@ end
     end
 end
 
+# This is equivalent to
+# extrema(_check_bitset_bounds, v)
+# but extrema(::Function, ::AbstractVector) is not available to the compiler
+function _bitset_extrema(itr)
+    x, i = iterate(itr)
+    lo = hi = _check_bitset_bounds(x)
+    while true
+        xi = iterate(itr, i)
+        xi === nothing && break
+        x, i = xi
+        y = _check_bitset_bounds(@inbounds x)
+        lo = min(lo, y)
+        hi = max(hi, y)
+    end
+    lo, hi
+end
+
+# assumes that x falls within the allocated bits of s
+function unsafe_push!(s::BitSet, x)
+    intx = Int(x)
+    i1 = _div64(intx) - s.offset + 1
+    i2 = _mod64(intx)
+    @inbounds s.bits[i1] |= 1 << i2
+    s
+end
+
+# assumes that every element of itr falls within the allocated bits of s
+function unsafe_union!(s::BitSet, itr)
+    for x in itr
+        unsafe_push!(s, x)
+    end
+    s
+end
+
+function BitSet(v::AbstractVector)
```

Comment:
It seems this will catch `BitSet(::AbstractUnitRange{<:Integer}`, which should still use the specialized `union!` method.